packages/ts/file-router/src/vite-plugin.ts

```diff
@@ -102,26 +102,39 @@ export default function vitePluginFileSystemRouter({
       server.watcher.on('unlink', changeListener);
     },
     transform(code, id): Promise<TransformResult> | TransformResult {
+      let modifiedCode = code;
       if (id.startsWith(fileURLToPath(_viewsDir)) && !basename(id).startsWith('_')) {
-        // To enable HMR for route files with exported configurations, we need
-        // to address a limitation in `react-refresh`. This library requires
-        // strict equality (`===`) for non-component exports. However, the
-        // dynamic nature of HMR makes maintaining this equality between object
-        // literals challenging.
-        //
-        // To work around this, we implement a strategy that preserves the
-        // reference to the original configuration object (`currentExports.config`),
-        // replacing any newly created configuration objects (`nextExports.config`)
-        // with it. This ensures that the HMR mechanism perceives the
-        // configuration as unchanged.
-        return {
-          code: code.replace(
+        if (isDevMode) {
+          // To enable HMR for route files with exported configurations, we need
+          // to address a limitation in `react-refresh`. This library requires
+          // strict equality (`===`) for non-component exports. However, the
+          // dynamic nature of HMR makes maintaining this equality between object
+          // literals challenging.
+          //
+          // To work around this, we implement a strategy that preserves the
+          // reference to the original configuration object (`currentExports.config`),
+          // replacing any newly created configuration objects (`nextExports.config`)
+          // with it. This ensures that the HMR mechanism perceives the
+          // configuration as unchanged.
+          modifiedCode = modifiedCode.replace(
             hmrInjectionPattern,
             `if (!nextExports) return;
-      if (Object.keys(nextExports).length === 2 && 'default' in nextExports && 'config' in nextExports) {
-        nextExports = { ...nextExports, config: currentExports.config };
-      }`,
-          ),
+            if (Object.keys(nextExports).length === 2 && 'default' in nextExports && 'config' in nextExports) {
+              nextExports = { ...nextExports, config: currentExports.config };
+            }`,
+          );
+        } else {
+          // In production mode, the function name is assigned as name to the function itself to avoid minification
+          const functionNames = /export\s+default\s+function\s+(\w+)/u.exec(modifiedCode);
```

Comment:
Here could be a problem. Sometimes, the user would like to use `const`, so the export could look like this:
```ts
const MyComponent: FC = () => {
  // ...
};

export default MyComponent;
```
In that case, the current RegExp won't work.

There is also a kind of an edge case:
```ts
function MyComponent() {
  // ...
}

export {
  MyComponent as default,
}
```
I don't think if anyone would really use that kind of export, so we can skip it for now.

Reply:
Would it be enough to only support the `export default function X` form for now? @Lodin

Reply:
No. I know a lot of devs who like to use `const` (I was one of them for some time as well), and that would be pretty inconvenient and frustrating for them.

Reply:
I guess, this change should be enough to take the additional case into account:
```suggestion
          const functionNames = /export\s+default\s+(?:function\s+)?(\w+)/u.exec(modifiedCode);
```

Reply:
![image](https://github.com/vaadin/hilla/assets/8037479/61381bd3-1fc3-4876-b0e8-1ae8cd37e7eb)
It looks like to work as expected